src/test/java/org/tarantool/AbstractTarantoolConnectorIT.java

```diff
@@ -0,0 +1,254 @@
+        assertTrue(List.class.isAssignableFrom(res.getClass()));
+        List list = (List)res;
+        assertEquals(1, list.size());
+        assertNotNull(list.get(0));
+        assertTrue(List.class.isAssignableFrom(list.get(0).getClass()));
+        assertEquals(tuple, list.get(0));
+    }
+
+    protected TarantoolClient getClient() {
+        TarantoolClientConfig config = new TarantoolClientConfig();
+        config.username = username;
+        config.password = password;
+        config.initTimeoutMillis = 1000;
+        config.sharedBufferSize = 128;
+
+        return new TarantoolClientImpl(socketChannelProvider, config);
+    }
+
+    protected static TarantoolConsole getConsole() {
+        return TarantoolConsole.open(host, consolePort);
+    }
+
+    protected static TarantoolConsole getConsole(String instance) {
+        return TarantoolConsole.open(host, tntCtlPort, instance);
+    }
+
+    protected TarantoolConnection getConnection() {
+        Socket socket = new Socket();
+        try {
+            socket.connect(new InetSocketAddress(host, port));
+        } catch (IOException e) {
+            throw new RuntimeException("Test failed due to invalid environment.", e);
+        }
+        try {
+            return new TarantoolConnection(username, password, socket);
+        } catch (Exception e) {
+            try {
+                socket.close();
+            } catch (IOException ignored) {
+                // No-op.
+            }
+            throw new RuntimeException(e);
+        }
+    }
+
+    protected List<?> consoleSelect(String spaceName, Object key) {
+        StringBuilder sb = new StringBuilder("box.space.");
+        sb.append(spaceName);
+        sb.append(":select{");
+        if (List.class.isAssignableFrom(key.getClass())) {
+            List parts = (List)key;
+            for (int i = 0; i < parts.size(); i++) {
+                if (i != 0)
+                    sb.append(", ");
+                Object k = parts.get(i);
+                if (k.getClass().isAssignableFrom(String.class)) {
+                    sb.append('\'');
+                    sb.append(k);
+                    sb.append('\'');
+                } else {
+                    sb.append(k);
+                }
+            }
+        } else {
+            sb.append(key);
+        }
+        sb.append("}");
+        return console.eval(sb.toString());
+    }
+
+    protected void stopTarantool(String instance) {
+        tarantoolControl("stop", instance);
+    }
+
+    protected void startTarantool(String instance) {
+        tarantoolControl("start", instance);
+    }
+
+    /**
+     * Control the given tarantool instance via tarantoolctl utility network endpoint.
+     *
+     * @param command A tarantoolctl utility command.
+     * @param instanceName Name of tarantool instance to control.
+     */
+    private void tarantoolControl(String command, String instanceName) {
+        int timeout = RESTART_TIMEOUT;
```

Comment:
Why custom reconnection strategy is implemented here? It seems the default one (from README.md) will do the work, aren't?

Reply:
I misplaced the comment, indended to comment `TestSocketChannelProvider.get()` method. Now I understood contract of this method better: raising an exception from `get()` means we cannot (re)connect at all, so the method implementation from README is not suitable for testing reconnection. So I'm okay with `TestSocketChannelProvider` you provided.

I think it would be good to better describe `SocketChannelProvider.get()` contract in README and provide your reconnection strategy implementation as second example in addition to fail-fast one we have now in README. Should I file an issue for that?

Reply:
Thanks!